Core/src/org/sleuthkit/autopsy/datasourcesummary/datamodel/DataSourceInfoUtilities.java

```diff
@@ -133,9 +141,204 @@ static String getMetaFlagsContainsStatement(TSK_FS_META_FLAG_ENUM flag) {
         return "meta_flags & " + flag.getValue() + " > 0";
     }
 
+    /**
+     * Enum for specifying the sort order for getAttributes.
+     */
+    enum SortOrder {
+        DESCENDING,
+        ASCENDING
+    }
+
+    /**
+     * Returns a list of all artifacts of the given type that have an attribute
+     * of the given type sorted by given attribute type value. Artifacts that do
+     * not have the given attribute will not be included in the list.
+     *
+     * Sorting on attributes of type byte[] and JSON is not currently supported.
+     *
+     * @param skCase        SleuthkitCase instance.
+     * @param artifactType  Type of artifacts to sort.
+     * @param dataSource    Data Source that the artifact belongs to.
+     * @param attributeType Attribute type to sort by.
+     * @param sortOrder     Sort order of the attributes, either ascending or
+     *                      descending.
+     *
+     * @return A list of artifacts of type artifactType sorted by the attribute
+     *         of attributeType in the given sortOrder. If no artifacts are
+     *         found an empty list will be returned.
+     *
+     * @throws TskCoreException
+     */
+    static List<BlackboardArtifact> getArtifacts(SleuthkitCase skCase, BlackboardArtifact.Type artifactType, DataSource dataSource, BlackboardAttribute.Type attributeType, SortOrder sortOrder) throws TskCoreException {
+        return getArtifacts(skCase, artifactType, dataSource, attributeType, sortOrder, 0);
+    }
+
+    /**
+     * Return a list of artifacts that have been sorted by their attribute of
+     * attributeType. If an artifact of the given type does not have the given
+     * attribute it will not be included in the returned list.
+     *
+     * Sorting on attributes of type byte[] and JSON is not currently supported.
+     *
+     * @param skCase        SleuthkitCase instance.
+     * @param artifactType  Type of artifacts to sort.
+     * @param dataSource    Data Source that the artifact belongs to.
+     * @param attributeType Attribute type to sort by.
+     * @param sortOrder     Sort order of the attributes, either ascending or
+     *                      descending.
+     * @param maxCount      Maximum number of results to return. To return all
+     *                      values maxCount should be 0.
+     *
+     * @return A list of artifacts of type artifactType sorted by the attribute
+     *         of attributeType in the given sortOrder. If no artifacts are
+     *         found an empty list will be returned.
+     *
+     * @throws TskCoreException
+     */
+    static List<BlackboardArtifact> getArtifacts(SleuthkitCase skCase, BlackboardArtifact.Type artifactType, DataSource dataSource, BlackboardAttribute.Type attributeType, SortOrder sortOrder, int maxCount) throws TskCoreException {
+        if (maxCount < 0) {
+            throw new IllegalArgumentException("Invalid maxCount passed to getArtifacts, value must be at greater 0");
```

Comment:
I think this is supposed to be "Invalid maxCount passed to getArtifacts; value must be greater than or equal to 0."